src/MichiTokenizeRequestor.sol

```diff
@@ -0,0 +1,121 @@
+
+    /// @notice event emitted when a new collection is added
+    event CollectionApproved(address indexed collection);
+
+    /// @notice event emitted when a collection has been removed
+    event CollectionRemoved(address indexed collection);
+
+    /// @dev Constructor for MichiTokenizeRequestor contract
+    /// @param michiTBALocker_ address of the MichiTBALocker
+    /// @param michiWalletReceiptNFT_ address of the MichiWalletReceiptNFT
+    constructor(address michiTBALocker_, address michiWalletReceiptNFT_) {
+        michiTBALocker = michiTBALocker_;
+        michiWalletReceiptNFT = IMichiWalletReceiptNFT(michiWalletReceiptNFT_);
+    }
+
+    /// @dev Initiate tokenized points request
+    /// @param michiWalletAddress tba address that holds points to be tokenized
+    function createTokenizePointsRequest(address michiWalletAddress) external {
+        // get tba ownership nft
+        (, address tokenContract, uint256 tokenId) = AccountV3Upgradable(payable(michiWalletAddress)).token();
+
+        if (!approvedCollections[tokenContract]) revert UnapprovedCollection(tokenContract);
+
+        // verify nft owner is caller
+        if (IERC721(tokenContract).ownerOf(tokenId) != msg.sender) revert UnauthorizedCaller(msg.sender);
+
+        // transfer in michi wallet nft
+        IERC721(tokenContract).safeTransferFrom(msg.sender, michiTBALocker, tokenId);
+
+        // mint the user a receipt NFT of same index
+        // in case other projects (such as AVSs) airdrop tokens to YT holders
+        michiWalletReceiptNFT.mint(msg.sender, tokenId);
+
+        _requestIds.increment();
+        uint256 newRequestId = _requestIds.current();
+        idToRequest[newRequestId] = Request(msg.sender, michiWalletAddress, newRequestId);
+
+        emit NewTokenizeRequest(msg.sender, michiWalletAddress, newRequestId);
+    }
+
+    /// @dev Add new collection
+    /// @param newCollection address of the new collection
+    function addApprovedCollection(address newCollection) external onlyOwner {
+        if (approvedCollections[newCollection]) revert CollectionAlreadyApproved(newCollection);
+        approvedCollections[newCollection] = true;
+
+        emit CollectionApproved(newCollection);
+    }
+
+    /// @dev Remove an approved collection
+    /// @param collectionToRemove address of the collection to remove
+    function removeApprovedCollection(address collectionToRemove) external onlyOwner {
+        if (!approvedCollections[collectionToRemove]) revert CollectionNotApproved(collectionToRemove);
+        approvedCollections[collectionToRemove] = false;
+
+        emit CollectionRemoved(collectionToRemove);
+    }
+
+    /// @dev Set a new MichiTBALocker address
+    /// @param newMichiTBALockerAddress address of the new MichiTBALocker
+    function setMichiTBALockerAddress(address newMichiTBALockerAddress) external onlyOwner {
+        michiTBALocker = newMichiTBALockerAddress;
```

Comment:
might want to validate that new address is not zero address and not the old address. Also good idea to emit event.

Reply:
added